### PR TITLE
MAGECLOUD-457: Add contribution guidelines

### PR DESCRIPTION
**Pull request by meker12**
### Description
Added contribution guidelines for the Magento Cloud Docker project

https://magento2.atlassian.net/browse/MAGECLOUD-4572

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+3. Commits must be accompanied by meaningful commit messages.
+4. For PRs which include bug fixes include a step-by-step description of how to reproduce the bug in the pull request descroption.
+3. For PRs which include new logic or new features, you must also submit the following information along with the pull request
+   * Unit/integration test coverage
+   * Proposed [documentation](https://devdocs.magento.com) updates: Submit developer documentation contributions from the [Magento DevDocs](https://github.com/magento/devdocs) repository. Submit updates to Magento user documentation from the [Magento Merchant documentation](https://github.com/magento/merchdocs) repository.
+4. For larger features or changes, [open an issue](https://github.com/magento/adobe-stock-integration/issues) to discuss the proposed changes prior to development. Discussing the updates in advance can prevent duplicate or unnecessary effort and allow other contributors to provide input.
```
Comment (bbatsche) [inline .github/CONTRIBUTING.md]: Need to update the URL here

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Contribution process
+1. Search current [listed issues](https://github.com/magento/adobe-stock-integration/issues) (open or closed) for similar proposals of intended contribution before starting work on a new contribution.
```
Comment (bbatsche) [inline .github/CONTRIBUTING.md]: Also URL here

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+5. After you submit the pull request, the Magento Cloud development team will review the contribution and collaborate with you as needed to incorporate your proposed changes.
+
+## Code of Conduct
+
+This project is released with a Contributor Code of Conduct. We expect you to agree to its terms when participating in this project.
+The full text is available in the repository [Wiki](https://github.com/magento/magento2/wiki/Magento-Code-of-Conduct).
+
+## Connecting with Community!
+
+If you have any questions, join us in [#adobe-stock](https://magentocommeng.slack.com/messages/CH8BGFX9D) Slack chat. If you are not on our slack, [click here](http://tinyurl.com/engcom-slack) to join.
```
Comment (bbatsche) [inline .github/CONTRIBUTING.md]: Should be #cloud-docker, and maybe check where the URL goes

**Review by bbatsche — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (YPyltiai) [reply .github/CONTRIBUTING.md]: URL is wrong I think. We can just refer to absolute URL of the Slack space.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Contributing to Magento Cloud Docker code
+
+Use the GitHub fork & pull model contribution model to submit your code contributions to the Magento Cloud Docker codebase.
+In this contribution model, you maintain your own [fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks) of the Magento Cloud Docker repository, and create a [pull request](https://help.github.com/articles/about-pull-requests/) to submit your proposed changes to the base repository. For details on the fork & pull contribution model, see the [Beginners guide](https://github.com/magento/magento2/wiki/Getting-Started).
+
+Contributions can take the form of new components or features, changes to existing features, tests, bug fixes, or optimizations. You can also contribute new or updated documentation.
```
Comment (YPyltiai) [inline .github/CONTRIBUTING.md]: Remove `components`. It's referring to base Magento & not applicable here.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+The Magento Cloud development team and community maintainers review all issues and contributions submitted by the developer community in first in, first out order (FIFO). During the review process, reviewers might notify a contributor to request clarification on the proposed changes. You have two weeks to respond to these requests. If we do not recieve a response, we close the pull request.
```
Comment (YPyltiai) [inline .github/CONTRIBUTING.md]: `You have two weeks to respond to these requests. If we do not recieve a response, we close the pull request.` - not sure if we need this sentence. At this point there are not that many PRs, so we can wait.

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+## Prerequisites
+
+You must have a [GitHub account](https://help.github.com/en/github/getting-started-with-github/signing-up-for-a-new-github-account) with [two-factor authentication](https://help.github.com/en/github/authenticating-to-github/configuring-two-factor-authentication) enabled to contribute to Magento repositories. We also recommend creating a [personal access token](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line) to use when interacting with GitHub in script from the the command line. 
+
+## Contribution requirements
+
+1. Contributions must adhere to the [Magento coding standards](https://devdocs.magento.com/guides/v2.3/coding-standards/bk-coding-standards.html).
```
Comment (YPyltiai) [inline .github/CONTRIBUTING.md:16]: @shiftedreality does it apply here?

**Review by YPyltiai — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (shiftedreality) [reply .github/CONTRIBUTING.md:16]: Yes

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (meker12) [inline .github/CONTRIBUTING.md]:
```suggestion
The Magento Cloud development team and community maintainers review all issues and contributions submitted by the developer community in first in, first out order (FIFO). During the review process, reviewers might notify a contributor to request clarification on the proposed changes.
```

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (meker12) [inline .github/CONTRIBUTING.md]:
```suggestion
Contributions can take the form of new features, changes to existing features, tests, bug fixes, or optimizations. You can also contribute new or updated documentation.
```

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (meker12) [reply .github/CONTRIBUTING.md]: I updated the URL to open the cloud-docker channel, and also changed #adobe-stock to #cloud-docker

**Review by shiftedreality — APPROVED** [review]

**Review by shiftedreality — DISMISSED** [review]

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+2. When you submit a Pull request (PR), write a meaningful description to explain the purpose of your contribution. Comprehensive descriptions increase the chances that a pull request can be merged quickly, without requests for additional clarification. See the [Magento Cloud Docker Pull Request Template](https://github.com/magento/magento-cloud-docker/blob/develop/.github/PULL_REQUEST_TEMPLATE.md) for more information.
+3. Commits must be accompanied by meaningful commit messages.
+4. For PRs which include bug fixes include a step-by-step description of how to reproduce the bug in the pull request descroption.
```
Comment (bbatsche) [inline .github/CONTRIBUTING.md]: "For PRs which include bug fixes, you must include a step-by-step description of how to reproduce the bug in the pull request description."

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+3. For PRs which include new logic or new features, you must also submit the following information along with the pull request
+   * Unit/integration test coverage
+   * Proposed [documentation](https://devdocs.magento.com) updates: Submit developer documentation contributions from the [Magento DevDocs](https://github.com/magento/devdocs) repository. Submit updates to Magento user documentation from the [Magento Merchant documentation](https://github.com/magento/merchdocs) repository.
```
Comment (bbatsche) [inline .github/CONTRIBUTING.md]: "Proposed [documentation](https://devdocs.magento.com) updates: Submit developer documentation contributions to the [Magento DevDocs](https://github.com/magento/devdocs) repository. Submit updates to Magento user documentation to the [Magento Merchant documentation](https://github.com/magento/merchdocs) repository."

**Review by bbatsche — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 7 above)
Comment (meker12) [inline .github/CONTRIBUTING.md]:
```suggestion
4. If your PR includes bug fixes, provide a step-by-step description of how to reproduce the bug in the pull request description.
```

File: .github/CONTRIBUTING.md (hunk 8 above)
Comment (meker12) [inline .github/CONTRIBUTING.md]:
```suggestion
   * Proposed documentation updates: Submit developer documentation contributions to the Magento DevDocs repository. Submit updates to Magento user documentation to the Magento Merchant documentation repository.
```

**Review by meker12 — COMMENTED** [review]
Added recommended revisions using GitHub suggestion feature.

**Review by shiftedreality — DISMISSED** [review]

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+## Contribution requirements
+
+1. Contributions must adhere to the [Magento coding standards](https://devdocs.magento.com/guides/v2.3/coding-standards/bk-coding-standards.html).
+2. When you submit a Pull request (PR), write a meaningful description to explain the purpose of your contribution. Comprehensive descriptions increase the chances that a pull request can be merged quickly, without requests for additional clarification. See the [Magento Cloud Docker Pull Request Template](https://github.com/magento/magento-cloud-docker/blob/develop/.github/PULL_REQUEST_TEMPLATE.md) for more information.
+3. Commits must be accompanied by meaningful commit messages.
+4. If your PR includes bug fixes, provide a step-by-step description of how to reproduce the bug in the pull request description.
+3. For PRs which include new logic or new features, you must also submit the following information along with the pull request
+   * Unit/integration test coverage
+   * Proposed documentation updates: Submit developer documentation contributions to the Magento DevDocs repository. Submit updates to Magento user documentation to the Magento Merchant documentation repository.
```
Comment (meker12) [inline .github/CONTRIBUTING.md]:
```suggestion
   * Proposed documentation updates: Submit developer documentation contributions to the [Magento DevDocs repository](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md). Submit updates to Magento user documentation to the [Magento Merchant documentation repository](https://github.com/magento/merchdocs/blob/master/.github/CONTRIBUTING.md).
```

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+## Contribution requirements
+
+1. Contributions must adhere to the [Magento coding standards](https://devdocs.magento.com/guides/v2.3/coding-standards/bk-coding-standards.html).
+2. When you submit a Pull request (PR), write a meaningful description to explain the purpose of your contribution. Comprehensive descriptions increase the chances that a pull request can be merged quickly, without requests for additional clarification. See the [Magento Cloud Docker Pull Request Template](https://github.com/magento/magento-cloud-docker/blob/develop/.github/PULL_REQUEST_TEMPLATE.md) for more information.
+3. Commits must be accompanied by meaningful commit messages.
+4. If your PR includes bug fixes, provide a step-by-step description of how to reproduce the bug in the pull request description.
+3. For PRs which include new logic or new features, you must also submit the following information along with the pull request
```
Comment (meker12) [inline .github/CONTRIBUTING.md]:
```suggestion
3. If your PR includes new logic or new features, you must also submit the following information along with the pull request
```

**Review by shiftedreality — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]

**Review by YPyltiai — APPROVED** [review]

**Pull request merged**